src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -185,6 +189,18 @@ private StatementSyntax Transform( StatementSyntax stmt )
 			_ => UnhandledSyntax( stmt )
 		};
 
+	private StatementSyntax Transform( ReturnStatementSyntax returnStmt ) {
+		var expr = returnStmt.Expression;
+		if( m_generatedFunctionReturnsVoid && expr is not null ) {
+			if( expr is InvocationExpressionSyntax || expr is AssignmentExpressionSyntax || expr is IIncrementOrDecrementOperation || expr is DeclarationExpressionSyntax || expr is MemberAccessExpressionSyntax ) {
+				return SyntaxFactory.ParseStatement( $"{Transform(expr).WithTriviaFrom(returnStmt)};\nreturn;" );

```

Comment:
Given we're just inserting a newline here and not applying other trivia, won't this end up like

```csharp
    return BarAsync();
  }
```
-->
```csharp
    Bar();
return;}
```

Reply:
I'm also kind of wondering what the output of ParseStatement is for this, like the type? Because I see two statements here?

We could do `SyntaxFactory.Block( SyntaxFactory.ExpressionStatement( Transform( expr ) ), SyntaxFactory.Return() );` (ignoring whitespace anyway). I worry that ParseStatement is coming up with a crazy answer that happens to just serialize back to string correctly here.

Reply:
When I put `return; return;` into roslyn-quoter and ask it to parse it as a statement I get back out:

```
ReturnStatement()
.WithSemicolonToken(
    Token(
        TriviaList(),
        SyntaxKind.SemicolonToken,
        TriviaList(
            new []{
                Trivia(
                    SkippedTokensTrivia()
                    .WithTokens(
                        TokenList(
                            Token(SyntaxKind.ReturnKeyword)))),
                Trivia(
                    SkippedTokensTrivia()
                    .WithTokens(
                        TokenList(
                            Token(SyntaxKind.SemicolonToken))))})))
.NormalizeWhitespace()
```

Like the extra return gets jammed into the trivia of the first one as a skipped token. This wouldn't compile but if you `.ToString()` it (like we do in this generator) it will round trip...

Reply:
I see... sorry I didn't notice this!
```
Block(ExpressionStatement(
                        InvocationExpression(
                            IdentifierName("Baz"))),
         ReturnStatement())))))
```
Definitely seems to be the correct way. I'll try to do that and maybe in the process fix the whitespace

Reply:
Looks like SyntaxFactory.Block() is kind of the only way. Normally we might worry about it causing scope issues because of the extra {} but since we're returning it shouldn't usually matter. It doesn't look the greatest but it's the only way with our generation model and apparently common in code generation